bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/exceptions/AuthSalusApiException.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.salus.internal.rest.exceptions;
+
+import java.io.Serial;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@SuppressWarnings("SerializableHasSerializationMethods")
+@NonNullByDefault
+public class AuthSalusApiException extends SalusApiException {
+    @Serial
+    private static final long serialVersionUID = 1L;
```

Comment:
See other exception, can you generate a `serialVersionUID` and remove the suppression?

Reply:
suppression is not about `serialVersionUID` 

![obraz](https://github.com/openhab/openhab-addons/assets/2962846/ff8816a2-1110-4a7e-9b08-693561e183fc)

[Non-serializable class with readObject() or writeObject()](https://www.jetbrains.com/help/idea/list-of-java-inspections.html#serialization-issues)

Reply:
I don't get it. This should work without any warning or issue:
- Drop @SuppressWarnings("SerializableHasSerializationMethods")
- Drop @serial 
- Add `private static final long serialVersionUID = -898870205407059008L;`
This is a common setup that almost all bindings use, vscode does not show that issue in your screenshot when setup liek this.

Reply:
Let's do a step back.

`@SuppressWarnings("SerializableHasSerializationMethods")` suppresses warning about not having `readObject` and `writeObject` methods. Apparently all serializable objects should have it, but as I see in our repo no one have it. You probably are not seeing this warning, because you are using Eclipse and I'm using IntelliJ. If you want to check the Java inspections for `Serializable` in IntelliJ please visit [this](https://www.jetbrains.com/help/idea/list-of-java-inspections.html#threading-issues) page and look for _Serializable class without readObject() and writeObject()_

I can drop `@Serial` annotation, but I don't want to. `@Serial` is similar annotation as `@Override` - it's not requiredto use it, but it is nice. 

> In this quick tutorial, we’ll take a look at the new @Serial annotation introduced with Java 14.
> Similarly to @Override, this annotation is used in combination with the serial lint flag to perform compile-time checks for the [serialization-related members of a class](https://www.baeldung.com/java-serialization).

[source](https://www.baeldung.com/java-14-serial-annotation)

Reply:
I use vscode, and don't see that warning. However i want to stick to the used pattern in other bindings. If you think it should be different, you are free to open an issue and discuss a mass PR to adjust all other cases.

Reply:
done

Reply:
Add private static final long serialVersionUID = -898870205407059008L;

Reply:
Why you want me to put `-898870205407059008L` instead of `1L`? What is better in `-898870205407059008L`?

Reply:
Documentation might be a bit overwhelming, but this post might help: https://stackoverflow.com/a/285809

both vscode and eclipse will warn you, not sure about your ide. Hope this is helpful.

Reply:
I understand why `serialVersionUID` is required. The question is why use `-898870205407059008L` instead of `1L`. You are aware that both numbers will give you consistent `serialVersionUID` value?

 `Therefore, to guarantee a consistent serialVersionUID value across different java compiler implementations, a serializable class must declare an explicit serialVersionUID value. ` - you just need a value, doesn't matter what value as long as it's there

Reply:
_The purpose of the serialization version UID is to keep track of different versions of a class in order to perform valid serialization of objects.
The idea is to generate an ID that is unique to a certain version of an class, which is then changed when there are new details added to the class, such as a new field, which would affect the structure of the serialized object.
Always using the same ID, such as 1L means that in the future, if the class definition is changed which causes changes to the structure of the serialized object, there will be a good chance that problems arise when trying to deserialize an object._

In this context i do not foresee fatal issues, true. It is a best practice to generate an uid and while already adding an explicit value it is just a second or so to just generate it.

Reply:
No no no, wait wait wait. I did not said anything about changing the value.

In the future when i will change the class you would probably propose another generated value like `62340987234L` when I will change the value to simply `2L`.

Both our solutions works with Serializable preconditions, but my have some extra features:
1. it's easier 
2. UID also shows you the version of the class 1st, 2nd, 3rd
3. you are sure that you will never use same values

`best practice to generate an uid` - btw how do you know that I didn't generate `1L`? there is the same probability of generating `-898870205407059008L` as is `1L` 😁

Reply:
Ahhh... it was generated and became 1L...... :-o

Let's continue.

Reply:
![i-am-totally-9adc73d212](https://github.com/openhab/openhab-addons/assets/2962846/e49b6acd-35a0-41d0-b8e9-fc6625869a91)